# E-Commerce Sales Analysis — Full Jupyter Notebook Project

**Project:** E-Commerce Sales Analysis  
**Type:** Data Analytics / Exploratory Data Analysis (EDA)

This notebook demonstrates a complete analytics workflow using Python, Pandas, NumPy and Matplotlib.

## 1. Problem Statement

An e-commerce company wants to understand its sales performance. The analysis covers:
- Revenue and profit
- Category performance
- Monthly trends
- Regional performance
- Top products
- Payment methods
- Discount and profitability
- Correlation analysis
- Business recommendations

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("ecommerce_sales_dataset.csv", parse_dates=["Order_Date"])
df["Profit"] = df["Revenue"] - df["Cost"]
df["Profit_Margin_Pct"] = df["Profit"] / df["Revenue"] * 100

pd.set_option("display.max_columns", None)
pd.set_option("display.float_format", lambda x: f"{x:,.2f}")
df.head()

## 2. Data Understanding

In [ ]:
print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Missing values:", int(df.isnull().sum().sum()))
print("Duplicate rows:", int(df.duplicated().sum()))
print("\nData types:")
print(df.dtypes)

Rows: 180
Columns: 14
Missing values: 0
Duplicate rows: 0

Data types:
Order_ID                     object
Order_Date           datetime64[ns]
Category                     object
Product                      object
Region                       object
Units                         int64
Unit_Price                    int64
Discount_Pct                  int64
Revenue                     float64
Cost                        float64
Payment_Method               object
Profit                      float64
Profit_Margin_Pct           float64
Month                        object


## 3. Descriptive Statistics

In [ ]:
df.describe().T

## 4. Key Performance Indicators

In [ ]:
total_revenue = df["Revenue"].sum()
total_profit = df["Profit"].sum()
total_units = df["Units"].sum()
average_order_value = df["Revenue"].mean()
profit_margin = total_profit / total_revenue * 100

print(f"Total Revenue: ₹{total_revenue:,.2f}")
print(f"Total Profit: ₹{total_profit:,.2f}")
print(f"Units Sold: {total_units:,}")
print(f"Average Order Value: ₹{average_order_value:,.2f}")
print(f"Profit Margin: {profit_margin:.2f}%")

Total Revenue: ₹1,302,689.30
Total Profit: ₹437,574.04
Units Sold: 534
Average Order Value: ₹7,237.16
Profit Margin: 33.59%


## 5. Category Analysis

In [ ]:
category_summary = df.groupby("Category").agg(
    Revenue=("Revenue","sum"),
    Profit=("Profit","sum"),
    Units=("Units","sum")
).sort_values("Revenue", ascending=False)

category_summary["Profit_Margin_%"] = (
    category_summary["Profit"] / category_summary["Revenue"] * 100
)
category_summary.round(2)

In [ ]:
plt.figure(figsize=(9,5))
plt.bar(category_summary.index, category_summary["Revenue"])
plt.title("Revenue by Category")
plt.xlabel("Category")
plt.ylabel("Revenue")
plt.xticks(rotation=20)
plt.tight_layout()
plt.show()

## 6. Monthly Revenue Trend

In [ ]:
df["Month"] = df["Order_Date"].dt.to_period("M").astype(str)
monthly_revenue = df.groupby("Month")["Revenue"].sum()
monthly_revenue

In [ ]:
plt.figure(figsize=(9,5))
plt.plot(monthly_revenue.index, monthly_revenue.values, marker="o")
plt.title("Monthly Revenue Trend")
plt.xlabel("Month")
plt.ylabel("Revenue")
plt.xticks(rotation=30)
plt.grid(True, alpha=0.3)
plt.tight_layout()
plt.show()

## 7. Regional Analysis

In [ ]:
region_revenue = df.groupby("Region")["Revenue"].sum().sort_values(ascending=False)
region_revenue

In [ ]:
plt.figure(figsize=(8,5))
plt.bar(region_revenue.index, region_revenue.values)
plt.title("Revenue by Region")
plt.xlabel("Region")
plt.ylabel("Revenue")
plt.tight_layout()
plt.show()

## 8. Top 10 Products

In [ ]:
top_products = df.groupby("Product").agg(
    Revenue=("Revenue","sum"),
    Units=("Units","sum")
).sort_values("Revenue", ascending=False).head(10)
top_products

In [ ]:
plt.figure(figsize=(10,6))
plt.barh(top_products.index[::-1], top_products["Revenue"][::-1])
plt.title("Top 10 Products by Revenue")
plt.xlabel("Revenue")
plt.tight_layout()
plt.show()

## 9. Payment Method Analysis

In [ ]:
payment_revenue = df.groupby("Payment_Method")["Revenue"].sum().sort_values(ascending=False)
payment_revenue

## 10. Discount and Profitability

In [ ]:
discount_summary = df.groupby("Discount_Pct").agg(
    Revenue=("Revenue","sum"),
    Profit=("Profit","sum"),
    Orders=("Order_ID","count")
)
discount_summary["Profit_Margin_Pct"] = discount_summary["Profit"] / discount_summary["Revenue"] * 100
discount_summary.round(2)

In [ ]:
plt.figure(figsize=(8,5))
plt.scatter(df["Discount_Pct"], df["Profit"])
plt.title("Discount Percentage vs Profit")
plt.xlabel("Discount (%)")
plt.ylabel("Profit")
plt.grid(True, alpha=0.3)
plt.tight_layout()
plt.show()

## 11. Correlation Analysis

In [ ]:
numeric_cols = ["Units","Unit_Price","Discount_Pct","Revenue","Cost","Profit","Profit_Margin_Pct"]
correlation = df[numeric_cols].corr()
correlation.round(2)

In [ ]:
plt.figure(figsize=(9,7))
plt.imshow(correlation, aspect="auto")
plt.colorbar(label="Correlation")
plt.xticks(range(len(correlation.columns)), correlation.columns, rotation=45, ha="right")
plt.yticks(range(len(correlation.columns)), correlation.columns)
plt.title("Correlation Matrix")
plt.tight_layout()
plt.show()

## 12. Key Business Findings

- **Highest revenue category:** Electronics
- **Highest revenue region:** West
- **Highest revenue product:** Bluetooth Speaker
- **Highest revenue month:** 2026-02
- **Highest revenue payment method:** Cash on Delivery
- **Overall profit margin:** 33.59%

## 13. Recommendations

1. Focus inventory planning on high-revenue categories.
2. Strengthen marketing in strong-performing regions and investigate weaker regions.
3. Promote top products through bundles and cross-selling.
4. Monitor discount levels because discounts can affect profitability.
5. Support popular payment methods while retaining alternatives.
6. Use monthly trends to plan inventory and promotional campaigns.

## Conclusion

This project demonstrates a complete EDA workflow: data loading, validation, KPI calculation, aggregation, visualization, correlation analysis, insights and business recommendations.

The synthetic dataset can be replaced with real company transaction data without changing the overall analysis structure.